obspy/signal/cross_correlation.py

```diff
@@ -76,7 +76,44 @@ def _xcorr_slice(a, b, shift, domain='freq'):
     return c[mid - shift:mid + shift + len(c) % 2]
 
 
-def correlate(a, b, shift, demean=True, normalize=True, domain='freq'):
+def _normxcorr(a, b, shift, domain='freq'):
+    """
+    Normalized cross-correlation
+    """
+    mid = (len(a) + len(b) - 1) // 2
+    if shift is None:
+        shift = mid
+    # Normalize short window before correlation
+    b = (b - b.mean()) / (b.std() * len(b))
+    # We HAVE to pad in this case
+    a = np.pad(a, pad_width=len(b), mode='constant', constant_values=0)
```

Comment:
np.pad is new in numpy 1.7. Obspy supports numpy down to 1.6. You could use the _pad_zeros function and later we can delete this and use np.pad everywhere.

Reply:
:+1:  thanks! I did wonder why `np.pad` wasn't used, but will remove.